EventFilter/HcalRawToDigi/plugins/HcalDigisProducerGPU.cc

```diff
@@ -113,106 +94,128 @@ void HcalDigisProducerGPU::acquire(edm::Event const& event,
   // raii
   cms::cuda::ScopedContextAcquire ctx{event.streamID(), std::move(holder), cudaState_};
 
-  hf01_.clear();
-  hf5_.clear();
-  hf3_.clear();
 
   // event data
   edm::Handle<HBHEDigiCollection> hbheDigis;
   edm::Handle<QIE11DigiCollection> qie11Digis;
   event.getByToken(hbheDigiToken_, hbheDigis);
   event.getByToken(qie11DigiToken_, qie11Digis);
+  
+  // tmp host data
+  HostCollectionf01 hf01;
+  HostCollectionf5 hf5;
+  HostCollectionf3 hf3;
+
+  // init f5 collection
+  if (hbheDigis->size() > 0) {
+    auto const nsamples = (*hbheDigis)[0].size();
+    auto const stride = hcal::compute_stride<hcal::Flavor5>(nsamples);
+    hf5.stride = stride;
+    hf5.reserve(config_.maxChannelsF5HB);
+  
+    // flavor5 get device blobs
+    df5_.stride = stride;
+    df5_.data = cms::cuda::make_device_unique<uint16_t[]>(
+      config_.maxChannelsF5HB * stride, ctx.stream());
+    df5_.ids = cms::cuda::make_device_unique<uint32_t[]>(config_.maxChannelsF5HB, ctx.stream());
+    df5_.npresamples = cms::cuda::make_device_unique<uint8_t[]>(config_.maxChannelsF5HB, ctx.stream());
+  }
 
-  // flavor 0/1 get devie blobs
-  df01_.data = cms::cuda::make_device_unique<uint16_t[]>(
-      config_.maxChannelsF01HE * hcal::compute_stride<hcal::Flavor01>(config_.nsamplesF01HE), ctx.stream());
-  df01_.ids = cms::cuda::make_device_unique<uint32_t[]>(config_.maxChannelsF01HE, ctx.stream());
-
-  // flavor3 get device blobs
-  df3_.data = cms::cuda::make_device_unique<uint16_t[]>(
-      config_.maxChannelsF3HB * hcal::compute_stride<hcal::Flavor3>(config_.nsamplesF3HB), ctx.stream());
-  df3_.ids = cms::cuda::make_device_unique<uint32_t[]>(config_.maxChannelsF3HB, ctx.stream());
-
-  // flavor5 get device blobs
-  df5_.data = cms::cuda::make_device_unique<uint16_t[]>(
-      config_.maxChannelsF5HB * hcal::compute_stride<hcal::Flavor5>(config_.nsamplesF5HB), ctx.stream());
-  df5_.ids = cms::cuda::make_device_unique<uint32_t[]>(config_.maxChannelsF5HB, ctx.stream());
-  df5_.npresamples = cms::cuda::make_device_unique<uint8_t[]>(config_.maxChannelsF5HB, ctx.stream());
+  if (qie11Digis->size() > 0) {
+    auto const nsamples = qie11Digis->samples();
+    auto const stride01 = hcal::compute_stride<hcal::Flavor01>(nsamples);
+    auto const stride3 = hcal::compute_stride<hcal::Flavor3>(nsamples);
+
+    hf01.stride = stride01;
+    hf3.stride = stride3;
+    hf01.reserve(stride01);
+    hf3.reserve(stride3);
```

Comment:
why these are 
```c++
    hf01.reserve(stride01);
    hf3.reserve(stride3);
```
while above there is
```c++
    hf5.reserve(config_.maxChannelsF5HB);
```
?

Should these rather be
```suggestion
    hf01.reserve(config_.maxChannelsF01HE);
    hf3.reserve(config_.maxChannelsF3HB);
```
?

Reply:
yeah that was stupid... fixed